ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                    phMiraMonLayer->nMMLanguage = MM_CAT_LANGUAGE;
+                else if (EQUAL(pszLanguage, "SPA"))
+                    phMiraMonLayer->nMMLanguage = MM_SPA_LANGUAGE;
+                else
+                    phMiraMonLayer->nMMLanguage = MM_ENG_LANGUAGE;
+            }
+            else
+                phMiraMonLayer->nMMLanguage = MM_DEF_LANGUAGE;  // Default
+
+            if (phMiraMonLayer->nSRS_EPSG != 0)
+            {
+                m_poSRS = new OGRSpatialReference();
+                m_poSRS->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
+                if (m_poSRS->importFromEPSG(phMiraMonLayer->nSRS_EPSG) !=
+                    OGRERR_NONE)
+                {
+                    delete m_poSRS;
+                    m_poSRS = nullptr;
+                }
+            }
+
+            // If there is associated information
+            if (phMiraMonLayer->pMMBDXP)
+            {
+                if (!phMiraMonLayer->pMMBDXP->pfDataBase)
+                {
+                    if ((phMiraMonLayer->pMMBDXP->pfDataBase =
+                             fopen_function(phMiraMonLayer->pMMBDXP->szFileName,
+                                            "r")) == nullptr)
+                    {
+                        CPLDebug("MiraMon", "File '%s' cannot be opened.",
+                                 phMiraMonLayer->pMMBDXP->szFileName);
+                        bValidFile = false;
+                        return;
+                    }
+
+                    // First time we open the extended DBF we create an index
+                    // to fastly find all non geometrical features.
+                    phMiraMonLayer->pMultRecordIndex = MMCreateExtendedDBFIndex(
+                        phMiraMonLayer->pMMBDXP->pfDataBase,
+                        phMiraMonLayer->pMMBDXP->nRecords,
+                        phMiraMonLayer->pMMBDXP->FirstRecordOffset,
+                        phMiraMonLayer->pMMBDXP->BytesPerRecord,
+                        phMiraMonLayer->pMMBDXP
+                            ->pField[phMiraMonLayer->pMMBDXP->IdGraficField]
+                            .AcumulatedBytes,
+                        phMiraMonLayer->pMMBDXP
+                            ->pField[phMiraMonLayer->pMMBDXP->IdGraficField]
+                            .BytesPerField,
+                        &phMiraMonLayer->isListField, &phMiraMonLayer->nMaxN);
+
+                    // Creation of maximum number needed for processing
+                    // multiple records
+                    if (phMiraMonLayer->pMultRecordIndex)
+                    {
+                        papszValues = static_cast<char **>(
+                            CPLCalloc((size_t)phMiraMonLayer->nMaxN + 1,
+                                      sizeof(*papszValues)));
+
+                        padfValues = static_cast<double *>(
+                            CPLCalloc((size_t)phMiraMonLayer->nMaxN,
+                                      sizeof(*padfValues)));
```

Comment:
no risks that those allocations may fail ? If nMaxN something read from the file and that can be arbitarily large? If the allocation may fail, then VSICalloc() should be used instead and return checked for nullptr

Reply:
I think you done that at some point. Tell me if not.